src/redis_info_provider/info_poller.py

```diff
@@ -118,19 +135,33 @@ def _poll_shard(self, shard):
                     self.logger.debug('Polled shard %s', shard.id)
                     info['meta'] = {}
                     shard.info = info
-                    consecutive_failures = 0
+                    consecutive_redis_failures = 0; consecutive_general_failures = 0
                     gevent.sleep(shard.polling_interval())
             except redis.RedisError:
-                consecutive_failures += 1
-                if consecutive_failures < self._grace:
+                consecutive_redis_failures += 1
+                if consecutive_redis_failures < self._grace:
                     self.logger.debug(
                         'Redis error polling shard %s for %d consecutive times; still within grace period; will retry',
-                        shard.id, consecutive_failures
+                        shard.id, consecutive_redis_failures
                     )
                 else:
                     self.logger.warning(
                         'Redis error polling shard %s for %d consecutive times; will retry',
-                        shard.id, consecutive_failures
+                        shard.id, consecutive_redis_failures
                     )
                 gevent.sleep(1)  # Cool-off period
                 continue  # Retry
+            except GreenletExit as e:
+                self.logger.info("poller %s exiting..", shard.id)
+                raise e
```

Comment:
I think not using a bare raise modifies the traceback...
```suggestion
                raise
```